File: web-version-V2/web-version-V2-Backend/src/core/wg.ts
import type { ProcessedServer, ValidatedConfig } from '../types';

export const buildWgConfig = (server: ProcessedServer, pubKey: string, opts: ValidatedConfig): string => `[Interface]
PrivateKey=${opts.privateKey}
Address=10.5.0.2/16
DNS=${opts.dns}

[Peer]
PublicKey=${pubKey}
AllowedIPs=0.0.0.0/0,::/0
Endpoint=${opts.useStation ? server.station : server.hostname}:51820
PersistentKeepalive=${opts.keepalive}`;

File: web-version-V2/web-version-V2-Backend/internal/wg/wg.go
package wg

import (
	"strconv"
	"strings"

	"nordgen/internal/types"
)

func Build(server types.ProcessedServer, pubKey string, opts types.ValidatedConfig) string {
	var sb strings.Builder
	sb.Grow(512)

	endpoint := server.Hostname
	if opts.UseStation {
		endpoint = server.Station
	}

	sb.WriteString("[Interface]\nPrivateKey=")
	sb.WriteString(opts.PrivateKey)
	sb.WriteString("\nAddress=10.5.0.2/16\nDNS=")
	sb.WriteString(opts.DNS)
	sb.WriteString("\n\n[Peer]\nPublicKey=")
	sb.WriteString(pubKey)
	sb.WriteString("\nAllowedIPs=0.0.0.0/0,::/0\nEndpoint=")
	sb.WriteString(endpoint)
	sb.WriteString(":51820\nPersistentKeepalive=")
	sb.WriteString(strconv.Itoa(opts.KeepAlive))

	return sb.String()
}
